### ISSUE TITLE
Allow code to run on server startup (Highlighting Discussion #15341)

### ISSUE THREAD

**Issue by larsqa** [vercel/next.js#40900]
### Describe the feature you'd like to request

I've read on several discussion pages that Contributors don't actively look at discussions, so I'm raising [discussion #15341](https://github.com/vercel/next.js/discussions/15341) into a feature request.

Next.js being a backend and frontend framework, there should be a way to run initialization code once before starting the server.

Examples could be:
- initialize Databases (Migrations/Seed)
- Check for directory permissions
- Check for specific environment variables and/or enumerate their types

From [discussion #15341](https://github.com/vercel/next.js/discussions/15341), the usual response is to "use a custom server". However, it feels wrong to use a discouraged Next.js feature for a such a crucial task.

### Describe the solution you'd like

Add a way to run code once during
- `npm run dev`
- `npm run start`
without using a custom server.

### Describe alternatives you've considered

I've used @icyJoseph's [approach](https://github.com/vercel/next.js/discussions/15341#discussioncomment-2987699), which seems to work, but would love to have an opinion from the maintainers about long time implementation of this "workaround"

**Comment by balazsorban44** [comment]
Hi, thanks for sharing this!

We are monitoring the discussions, and try prioritizing feature requests based on community engagement as a best effort.

As stated in our [Contributing documentation](https://github.com/vercel/next.js/blob/canary/contributing/repository/triaging.md), feature requests are converted to discussions by default, so probably it's best to continue the discussion in the original thread #15341

**Comment by larsqa** [comment]
@balazsorban44 Thanks a lot. It would be nice to have a more official reply on this from a Next.js member, since it's been quite old now and users have just "guessed" solutions.
